phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -554,4 +551,99 @@ private boolean extractTablesAndIndexes(PhoenixConnection conn) {
             return false;
         }
     }
+
+    private void prepareToRebuildIndexes(Connection conn) {
+        try {
+            HashMap<String, IndexInfo> rebuildIndexes = new HashMap<>();
```

Comment:
This map needs to be inside the for loop below or be cleared in between iterations.

Reply:
Thank you for the catch, passing the data table parameter to avoid recomputations